src/systems/logical_audio_sensor_plugin/Source.hh

```diff
@@ -0,0 +1,245 @@
+                   const float _falloffDistance,
+                   const float _volumeLevel,
+                   const bool _playing);
+
+    /// \brief Constructor.
+    /// \param[in] _id The source's ID.
+    /// \param[in] _position The source's position (x,y,z).
+    /// \param[in] _attenuationFunc The attenuation function.
+    /// \param[in] _attenuationShape The attenuation shape.
+    /// \param[in] _innerRadius The inner radius.
+    /// \param[in] _falloffDistance The falloff distance.
+    /// \param[in] _volumeLevel The volume emitted from the source.
+    /// \param[in] _playing Whether the source is playing by default.
+    /// \sa Source() for more information about valid constructor parameters.
+    public: Source(const unsigned int _id,
+                   const ignition::math::Vector3d &_position,
+                   const std::string &_attenuationFunc,
+                   const std::string &_attenuationShape,
+                   const float _innerRadius,
+                   const float _falloffDistance,
+                   const float _volumeLevel,
+                   const bool _playing);
+
+    /// \brief Computes the volume level of the source at a specific location.
+    /// \param[in] _location The location where the volume level should be
+    ///   calculated.
+    /// \return The volume level at this location.
+    ///   If the attenuation function or shape is undefined, -1.0 is returned.
+    ///   If the source is not playing, 0.0 is returned.
+    public: float VolumeLevel(const ignition::math::Vector3d &_location) const;
+
+    /// \brief Start playing audio from the source. If the source is already
+    ///   playing audio, nothing will change (the source will keep playing).
+    public: void StartPlaying();
+
+    /// \brief Stop playing audio from the source. If the source has already
+    ///   been stopped, nothing will change (the source will remain stopped).
+    public: void StopPlaying();
+
+    /// \brief Check to see if the audio source is playing or not.
+    /// \return \c true if the source is playing, \c false otherwise.
+    public: bool IsPlaying() const;
+
+    /// \brief Set the source's position.
+    /// \param[in] _position The new position of the source.
+    public: void SetPosition(const ignition::math::Vector3d &_position);
+
+    /// \brief Set the attenuation function that matches the defined string.
+    ///   The string is case sensitive, and must match the spelling
+    ///   of the values in AttenuationFunction. If the spelling does not match,
+    ///   the attenuation function is set as Undefined.
+    ///
+    ///   \em Example: to set the attenuation function to
+    ///     AttenuationFunction::Linear, the following
+    ///     are examples of valid strings: "linear", "Linear", "LINEAR"
+    /// \param[in] _attenuationFunc a string that should map to a value in
+    ///   AttenuationFunction.
+    private: void SetAttenuationFunction(std::string _attenuationFunc);
```

Comment:
Prefer `const std::string &`

Reply:
I'm passing the strings in as a copy because I have to transform the strings to lowercase in order to do a case sensitive comparison:

https://github.com/ignitionrobotics/ign-gazebo/blob/e1689aef5d4966ae7a515f0877d685e8c49afb63/src/systems/logical_audio_sensor_plugin/LogicalAudio.cc#L103-L108

So, in this case, is it okay to leave the method signature as is (i.e., passing the strings as a copy)?